### ISSUE TITLE
SSR issue with core

### ISSUE THREAD

**Issue by vishalmanohar** [vmware/clarity#5597]
## Describe the bug

After upgrading to Angular 11, Clarity 5.1, SSR using angular universal is not working due a `window` reference in `lit-html` dependency

```
if (typeof window.ShadyCSS === 'undefined') {
^

ReferenceError: window is not defined
```

Dependency versions

```
    "@cds/core": "^5.0.0",
    "@clr/angular": "^5.0.1",
    "@clr/core": "^4.0.11",
    "@clr/icons": "^5.0.1",
    "@clr/ui": "^5.0.1",
    "@angular/common": "^11.1.2",
    "@angular/compiler": "^11.1.2",
    "@angular/compiler-cli": "^11.1.2",
    "@angular/core": "^11.1.2",
    "@angular/forms": "^11.1.2",
    "@angular/localize": "^11.1.2",
    "@angular/material": "^9.2.4",
    "@angular/platform-browser": "^11.1.2",
    "@angular/platform-browser-dynamic": "^11.1.2",
    "@angular/platform-server": "^11.1.2",
    "@angular/pwa": "0.6.8",
    "@angular/router": "^11.1.2",
    "@angular/service-worker": "^11.1.2",
```

## Expected behavior
No error on running SSR

## Versions

**Clarity project:**

- [ ] Clarity Core
- [x] Clarity Angular/UI

**Clarity version:**

- [ ] v3.x
- [ ] v4.x
- [x] v5.x

**Framework:**

- [x] Angular
- [ ] React
- [ ] Vue
- [ ] Other:

**Framework version:**
Angular 11

## Additional notes
Seems related to https://github.com/vmware/clarity/issues/4279
Kindly suggest a workaround if available.

**Comment by sis0k0** [comment]
I managed to reproduce this in a new Clarity project. 

```
ng new universal-test
cd universal-test
ng add @clr/angular
ng add @nguniversal/express-engine
npm run dev:ssr
```

**Comment by mathisscott** [comment]
Ping @coryrylan

**Comment by coryrylan** [comment]
Typically when using only `@cds/core`  it has to be imported in the browser module for SSR to work that way the custom element is only active in the browser. You can find an example in our Universal demo app here
https://github.com/vmware/clarity/tree/next/apps/core-angular-universal 

I'll dig into looking at what needs to be adjusted for the ClrModule in `clr-angular`, I'm thinking that might be the source issue.

**Comment by vishalmanohar** [comment]
@coryrylan Did you get a chance to look at this? Perhaps if you can point me to the area of code, I can submit a PR.

**Comment by arnifreyrs** [comment]
This is also a problem in cds/react version 5.0.3. We have a server side rendered app using next js, and we have to dynamically load all CDS-icons in the browser, because otherwise the SSR fails with the aforementioned error from `lit-html`

**Comment by vishal-vndly** [comment]
@coryrylan @mathisscott is there any workaround for this issue when using clr-angular?

**Comment by coryrylan** [comment]
Right now the best option is to load the Core components only in the Browser module like in [this demo](https://github.com/vmware/clarity/tree/next/apps/core-angular-universal). I will be updating this demo to the latest Core/Angular soon. There are some potential options in the future for loading on the SSR side but are not ready for production just yet.

**Comment by vishalmanohar** [comment]
Thanks @coryrylan

Tried loading the core components and icons used only in browser module. 
However was still getting the window error when using any clr/angular component like `clr-modal`

```
const policy = window.trustedTypes &&
               ^

ReferenceError: window is not defined
```

Created a repo [ng-universal-test](https://github.com/vishalmanohar/ng-universal-test) that reproduces this.

There seem to be imports on `cds/core` components in the `ClarityModule` and other other clarity angular components too.

https://github.com/vmware/clarity/blob/7e28669d57b006f72ac0b2dca8fd8e0b16044cf8/packages/angular/projects/clr-angular/src/clr-angular.module.ts#L29

Am I missing something?

**Comment by coryrylan** [comment]
@vishalmanohar thanks for the reproduction! It looks like this was a regression in the Clarity Module. I made a issue to fix the particular issue you found in #6042

**Comment by miguelchico** [comment]
ey @coryrylan, any progress on this or the related #6042 ?
This is a stopper for everybody wanting to use SSR and Clarity
